pkg/report/report.go

```diff
@@ -521,6 +517,72 @@ func Generate(ctx context.Context, path string, mrs yara.MatchRules, c malconten
 		// TODO: If we match multiple rules within a single namespace, merge matchstrings
 	}
 
+	// Handle overrides
+	// Create maps to store the original rule's behavior and rule behaviors that aren't the original
+	originalMap := make(map[string]*malcontent.Behavior)
+	overrideMap := make(map[string]*malcontent.Behavior)
+
+	// Store the override rule's details in its map and remove it from the slice of behaviors
+	// Add the remaining rules to the original map for verification
+	for i := 0; i < len(fr.Behaviors); {
+		b := fr.Behaviors[i]
+
+		// Check if this behavior is an override that needs to be removed
+		// If it is, store its behavior in the override map and mark it for deletion
+		isOverride := false
+		for _, vo := range validOverrides {
+			if b.RuleName == vo.Override {
+				overrideMap[vo.Override] = b
+				isOverride = true
+				break
+			}
+		}
+
+		// If this is the override rule, delete it from the behavior slice
+		// Otherwise, treat the rule as a possible original (vo.Original makes this an O(1) lookup)
+		if isOverride {
+			fr.Behaviors = slices.Delete(fr.Behaviors, i, i+1)
+		} else {
+			originalMap[b.RuleName] = b
+			i++
+		}
+	}
+
+	// Apply all valid overrides using populated maps for easy lookups
+	for _, vo := range validOverrides {
+		original, originalExists := originalMap[vo.Original]
+		_, overrideExists := overrideMap[vo.Override]
+
+		// If the original and override rules exist,
+		// update the original rule with the correct risk level and score
+		if originalExists && overrideExists {
+			for _, b := range fr.Behaviors {
+				if b.RuleName == original.RuleName {
+					b.RiskLevel = RiskLevels[vo.Severity]
+					b.RiskScore = vo.Severity
+					break
+				}
+			}
+		}
+	}
+
+	// Regardless of running an `analyze` or `scan`, adjust the overall risk if we deviated from overallRiskScore
+	// Scans will still need to drop medium or lower results
+	newRisk := highestBehaviorRisk(fr)
+	switch c.Scan {
+	case true:
+		switch {
+		case newRisk != overallRiskScore && newRisk >= 3:
```

Comment:
Can this use a const of some sort? I don't remember if "3" is "HIGH" or "CRITICAL". I'm not sure why either needs to be hardcoded here to be honest, but I think it'd improve readability.

Reply:
Updated in [`660024a` (#487)](https://github.com/chainguard-dev/malcontent/pull/487/commits/660024ab189cc7f5e1bc854b07ddd0c7d50510ba).